### ISSUE TITLE
Vscode - Cannot find module './App.svelte' or its corresponding type declarations.

### ISSUE THREAD

**Issue by Dibbyo456** [sveltejs/language-tools#1556]
### Describe the bug

When the extension installed after scaffolding a project if I open main.js, vs code complains about the error even though everything is perfectly working and valid.

### Reproduction

```
npm init vite my-app -- --template svelte
cd my-app
npm install
npm run dev
```

And open VS Code.

### Expected behaviour

Should not complain

### System Info

Windows 10
VS Code: 1.69.0
Svelte for VS Code v105.18.1

### Which package is the issue about?

Svelte for VS Code extension

### Additional Information, eg. Screenshots

![image](https://user-images.githubusercontent.com/18257605/178275630-da9626d8-ebd8-48d7-a53e-3a5189698ee3.png)

**Comment by dummdidumm** [comment]
Did you enable the typescript plugin (the svelte extension asks about enabling it)?

**Comment by Dibbyo456** [comment]
Nope I did not. I only installed the extension and nothing else. Like I said before I even tried on a fresh os, result are still same.

**Comment by dummdidumm** [comment]
Just to be clear: this setting https://github.com/sveltejs/language-tools/tree/master/packages/svelte-vscode#svelteenable-ts-plugin is false in your settings? What happens if you turn it on and then reload vs code?

**Comment by Dibbyo456** [comment]
Yes it false as I’m using JS (not TS)

**Comment by Dibbyo456** [comment]
I believe the extension somehow interpreting it as TypeScript which causing the errors.

**Comment by dummdidumm** [comment]
What happens if you turn on the plugin (it applies to JS files, too)?

**Comment by Dibbyo456** [comment]
Well altering setting `"svelte.enable-ts-plugin": true` the error disappeared. Could you explain the reason? Thanks!

**Comment by jasonlyu123** [comment]
The error is coming from typescript, which also powers the javascript language feature. It's normal to have this error for file extensions not supported by typescript. But it can be suppressed if you have a type definition for the file extensions. And the `vite-env.d.ts` is for this purpose. It just won't work until you have installed the dependencies. 

Not sure what happens in your system. But I can't reproduce this when I didn't enable the ts plugin. The error only shows up before `npm install`. Which is expected. If I open the editor first and then run `npm install`. The error would disappear after installing finished. Usually, even if the error doesn't disappear after installing. It should also disappear after restarting your editor.

The typescript plugin is another way to solve the problem by injecting logic into the tssever that powers the js/ts language features. To allow tssever to know how to process a svelte file. This feature also doesn't work before `npm install`. But this is because we won't start the plugin if we detected it's a svelte project, for performance reasons. 

TLDR. You can try opening the editor after you run `npm install`. Or restart your editor after `npm install`.

**Comment by dummdidumm** [comment]
My hunch was that maybe the "remove the ambient type definitions" logic which we added recently also runs when the plugin is disabled, which shouldn't happen

**Comment by Dibbyo456** [comment]
Reopened because it could actually be a bug.

**Comment by jasonlyu123** [comment]
Ah right. that's the problem. We indeed didn't check the config in the "remove the ambient type definitions" logic. Ironically what I said you can try is the actual reproduction of the problem. Sorry I didn't check it more.

**Comment by Coreusa** [comment]
Experienced the same with a fresh install with Svelte (Typescript) via Yarn 2. Adding this in `svelte.config.js` fixed the issue:

/** @type {import('@sveltejs/kit').Config} */
const config = {
	// Consult https://github.com/sveltejs/svelte-preprocess
	// for more information about preprocessors
	preprocess: preprocess(),
	"enable-ts-plugin": true, **<-- THIS ONE**
	kit: {
		adapter: adapter()
	}
};

The installation was fresh, so this is still a bug and not fixed.

**Comment by crimx** [comment]
It seems not working properly in a monorepo when multiple packages export their own `.svelte`s in `index.ts`. After reloading VSCode windows, only the active package (last focused editor editing the file within) is fine. Other packages' `index.ts` are showing the `Cannot find module './xxx.svelte' or its corresponding type declarations.` error.

Manually adding `xxx.svelte.d.ts` fixes the issue but not a very pleasant workflow.

**Comment by jwerre** [comment]
Thanks @jasonlyu123 for this:
> It should also disappear after restarting your editor.

I ran into this same issue and it went away after restarting VS Code.

**Comment by jarecsni** [comment]
Same here, still not fixed.

**Comment by RehkMansa** [comment]
Same issue with yarn installation and I have enable-ts-plugin set as true

**Comment by jwerre** [comment]
I'm getting this issue again, restart is not helping this time.

**Comment by dummdidumm** [comment]
Someone from the latest commenters, please provide a reproduction (ideally a repository) and environment info like the used vs code version.

**Comment by RehkMansa** [comment]
> Someone from the latest commenters, please provide a reproduction (ideally a repository) and environment info like the used vs code version.

https://github.com/RehkMansa/svelte-rebuild
vs code version: 1.74.2
extension: JavaScript and TypeScript, Svelte for VS Code

**Comment by dummdidumm** [comment]
Does the error go away if you disable the Svelte typescript plugin and restart vs Code?

**Comment by nbodinripert** [comment]
If I disable checkJs in jsconfig.json the error goes away

![Capture d’écran 2023-05-24 à 12 02 23](https://github.com/sveltejs/language-tools/assets/20435162/bac08c56-52b0-43a4-904a-aed2d66ae725)

**Comment by winston0410** [comment]
don't understand why @nbodinripert 's fix works, but it works for me

**Comment by akx** [comment]
My resolution for this: If your tsconfig.json `include` setting is

```
"include": ["src/**/*.d.ts", "src/**/*.js", "src/**/*.svelte"]
```
(which is the default that e.g. Vite's Svelte starter ships) you'll need to make it
```
"include": ["src/**/*.d.ts", "src/**/*.js", "src/**/*.ts", "src/**/*.svelte"]
```

**Comment by vinavinas** [comment]
Create a type file and declare module (just like shims-vue.d.ts in vue)
```ts
// inside ./src/svelte.d.ts
declare module '*.svelte'
```

**Comment by LeonAvancini** [comment]
> Create a type file and declare module (just like shims-vue.d.ts in vue)
> 
> ```ts
> // inside ./src/svelte.d.ts
> declare module '*.svelte'
> ```

This works for me, can you explain me a under the hood what this does and how this fix the problem? just a brief explanation will be more than enough, thanks before hand :)
